OpenRA.Mods.Cnc/Traits/Render/RenderShroudCircle.cs

```diff
@@ -19,37 +19,60 @@
 
 namespace OpenRA.Mods.Cnc.Traits
 {
-	class RenderShroudCircleInfo : TraitInfo<RenderShroudCircle>, IPlaceBuildingDecorationInfo
+	class RenderShroudCircleInfo : ITraitInfo, IPlaceBuildingDecorationInfo
 	{
+		[Desc("Color of the circle.")]
+		public readonly Color Color = Color.FromArgb(128, Color.Cyan);
+
+		[Desc("Contrast color of the circle.")]
+		public readonly Color ContrastColor = Color.FromArgb(96, Color.Black);
+
 		public IEnumerable<IRenderable> Render(WorldRenderer wr, World w, ActorInfo ai, WPos centerPosition)
 		{
 			var localRange = new RangeCircleRenderable(
 				centerPosition,
 				ai.TraitInfo<CreatesShroudInfo>().Range,
 				0,
-				Color.FromArgb(128, Color.Cyan),
-				Color.FromArgb(96, Color.Black));
+				Color,
+				ContrastColor);
 
 			var otherRanges = w.ActorsWithTrait<RenderShroudCircle>()
 				.SelectMany(a => a.Trait.RangeCircleRenderables(a.Actor, wr));
 
 			return otherRanges.Append(localRange);
 		}
+
+		public object Create(ActorInitializer init) { return new RenderShroudCircle(init.Self, this); }
 	}
 
 	class RenderShroudCircle : IRenderAboveShroudWhenSelected
 	{
+		readonly RenderShroudCircleInfo info;
+
+		public RenderShroudCircle(Actor self, RenderShroudCircleInfo info)
+		{
+			this.info = info;
+		}
+
 		public IEnumerable<IRenderable> RangeCircleRenderables(Actor self, WorldRenderer wr)
 		{
 			if (!self.Owner.IsAlliedWith(self.World.RenderPlayer))
 				yield break;
 
+			var range = self.TraitsImplementing<CreatesShroud>()
```

Comment:
This can be cached in the ctor.

Reply:
Not in the ctor, since there is no Requires<>, but in INotifyCreated.